pkg/utils/whisper/whisper_test.go

```diff
@@ -0,0 +1,94 @@
+package whisper_test
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	api "github.com/rotationalio/whisper/pkg/api/v1"
+	"github.com/stretchr/testify/require"
+	"github.com/trisacrypto/directory/pkg/utils/whisper"
+)
+
+func TestCreateWhisperLink(t *testing.T) {
+	// CreateWhisperLink should throw an error when passed an empty secret
+	accesses := 3
+	oneWeek := time.Now().AddDate(0, 0, 7)
+	link, err := whisper.CreateSecretLink("", "not empty", accesses, oneWeek)
+	require.Equal(t, link, "")
+	require.EqualError(t, err, "a secret is required to generate a Whisper link")
+
+	fixture := &api.CreateSecretReply{
+		Token:   "abcdefghijklmnop",
+		Expires: oneWeek,
+	}
+
+	// Create a Test Server
+	testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, r.Method, http.MethodPost)
+		require.Equal(t, "/v1/secrets", r.URL.Path)
+
+		in := new(api.CreateSecretRequest)
+		err := json.NewDecoder(r.Body).Decode(in)
+		require.NoError(t, err)
+
+		require.Equal(t, in.Accesses, 3)
+		require.NotNil(t, in.Lifetime)
+
+		w.Header().Add("Content-Type", "application/json; charset=utf-8")
+		w.WriteHeader(http.StatusCreated)
+		json.NewEncoder(w).Encode(fixture)
+	}))
+	defer testServer.Close()
+
+	// Create a Client that makes requests to the test server
+	err = whisper.ConnectClient(testServer.URL)
+	defer whisper.ResetClient()
+	require.NoError(t, err)
+
+	// Call CreateSecretLink. The call should perform a Post to the test server.
+	link, err = whisper.CreateSecretLink("this is a secret", "password", accesses, oneWeek)
+	require.NoError(t, err)
+	require.Equal(t, link, "https://whisper.rotational.dev/secret/abcdefghijklmnop")
+}
+
+// Live test using the Whisper server to ensure that we can fetch the secret using the
+// created Whisper link and that the correct number of accesses has been set. You must
+// set the GDS_TEST_LIVE_WHISPER environment variable equal to '1' in order to run this
+// test.
+func TestCreateWhisperLinkLive(t *testing.T) {
+	if os.Getenv("GDS_TEST_LIVE_WHISPER") != "1" {
```

Comment:
This is a really small thing but can you change this to accept any string? I'm worried about our future selves trying to test this and forgetting that we need to set it to exactly "1".

Reply:
Updated